### PR TITLE
Parrable context

### PR DESCRIPTION
**Pull request by chantalgo**
Patch for the Parrable context according to the schema change on #404

### CODE REVIEW

**Comment by chantalgo** [comment]
@alexanderdean Could you help me figure out the issue here? Does it have to do with the fact that the schema change has not yet been patched?

File: src/js/tracker.js [hunk 1]
```diff
@@ -1087,6 +1095,24 @@
...
 		/**
+		 * Creates a context from the window['_hawk'] object
+		 *
+		 * @return object The Parrable context
+		 */
+		function getParrableContext() {
+			var parrable = window['_hawk'];
+			if (parrable) {
+				var context = { encryptedID: null, optout: null};
+				context['encryptedID'] =  parrable.browserid;
```
Comment (alexanderdean) [inline src/js/tracker.js]: I think it's `encryptedId`, not `ID`

**Comment by alexanderdean** [comment]
Yes @chantalgo the schema change hasn't been released yet...

**Comment by chantalgo** [comment]
We usually serve the output value inside our async callback `_hawk("parrable", function(parrable) { … }` but that requires being called after we can assert `hawk.js` has been loaded. 

I added a commit that should be a fix for the output value.

It is also important to note that since we are using the variable`browserid` (readily available upon load) instead of the value obtained from the callback, the first impression value is lost and it can be null. (Using the callback would guarantee a value.)

**Comment by alexanderdean** [comment]
> It is also important to note that since we are using the variablebrowserid (readily available upon load) instead of the value obtained from the callback, the first impression value is lost and it can be null. (Using the callback would guarantee a value.)

Yes, that's similar to our Augur implementation. We have a ticket to add a dependency system for inter-async-JS dependencies (#401) but that hasn't been actioned yet.

**Comment by chantalgo** [comment]
@alexanderdean Do I need to make any changes for the updates?

**Comment by alexanderdean** [comment]
Hi @chantalgo - no, you should be good to go!

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: **Please visit https://github.com/snowplow/snowplow/wiki/CLA to learn more and sign.**

Once you've signed, please reply here (e.g. `I signed it!`) and we'll verify. Thanks.

File: src/js/tracker.js [hunk 2]
```diff
@@ -1087,6 +1095,25 @@
...
+		 *
+		 * @return object The Parrable context
+		 */
+		function getParrableContext() {
+			var parrable = window['_hawk'];
+			if (parrable) {
+				var context = { encryptedId: null, optout: null};
+				context['encryptedId'] =  parrable.browserid;
+				var regex = new RegExp('(?:^|;)\\s?' + "_parrable_hawk_optout".replace(/([.*+?^=!:${}()|[\]\/\\])/g, '\\$1') + '=(.*?)(?:;|$)', 'i'), match = document.cookie.match(regex);
+				parrable['optout'] = (match && decodeURIComponent(match[1])) ? match && decodeURIComponent(match[1]) : "false";
```
Comment (alexanderdean) [inline src/js/tracker.js]: Do you mean `context['optout'] = ...` here?

Comment (chantalgo) [inline src/js/tracker.js]: Yes, apologies

**Comment by chantalgo** [comment]
Any issues on why it fails, please let me know so I can fix, thank you  @alexanderdean  for your time

**Comment by chantalgo** [comment]
CLA has been signed for the Parrable tem

**Comment by snowplowcla** [comment]
Confirmed! @chantalgo has signed the Software Grant and Corporate Contributor License Agreement. Thanks so much.

**Comment by alexanderdean** [comment]
LGTM! Scheduling into 2.7.0 assuming the rest of the Parrable integration can be completed by then.

**Comment by chuwy** [comment]
Closing in favor #525. Thanks.

**Pull request closed without merging**